### PR TITLE
Custom role definitions for MHSM

### PR DESCRIPTION
**Pull request by daviddesberg**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json [hunk 1]
```diff
@@ -27,38 +27,38 @@
     "application/json"
   ],
   "paths": {
-    "/{scope}/providers/Microsoft.Authorization/roleDefinitions": {
-      "get": {
+        "/{scope}/providers/Microsoft.Authorization/roleDefinitions/{roleDefinitionName}": {
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json:30]: Nit: extraneous whitespace.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json [hunk 2]
```diff
@@ -27,38 +27,38 @@
...
+      "delete": {
         "tags": [
           "RoleDefinitions"
         ],
-        "operationId": "RoleDefinitions_List",
-        "description": "Get all role definitions that are applicable at scope and above.",
+        "operationId": "RoleDefinitions_Delete",
+        "description": "Deletes a custom role definition.",
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json:36]: Question: should an error be expected if someone tries to delete a built-in role definition?

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json [hunk 3]
```diff
@@ -68,47 +68,50 @@
...
             "$ref": "#/parameters/ApiVersionParameter"
           }
         ],
         "responses": {
-          "200": {
-            "description": "OK - Returns information about the role assignment.",
+          "201": {
+            "description": "Created - Returns information about the role definition.",
             "schema": {
               "$ref": "#/definitions/RoleAssignment"
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json:116]: This should return a RoleDefinition.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json [hunk 4]
```diff
@@ -68,47 +68,50 @@
...
-          "ListRoleDefinitions": {
-            "$ref": "./examples/ListRoleDefinitions-example.json"
+          "DeleteRoleDefinition": {
+            "$ref": "./examples/DeleteRoleDefinition-example.json"
           }
-        },
-        "x-ms-odata": "#/definitions/RoleDefinitionFilter"
-      }
-    },
-    "/{scope}/providers/Microsoft.Authorization/roleAssignments/{roleAssignmentName}": {
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json:82]: It seems you deleted all the role assignment CRUD operations. Looking at the file sans diff, I don't see them anywhere. These still need to be supported. I.e., I expect this file to contain only additions for custom role definitions.

**Review by heaths — CHANGES_REQUESTED** [review]
This seems to have replaced custom role assignments (necessary to assign principals to definitions) with custom role definitions when, in fact, custom role definitions should be additive, yes? In that case, you need to add (custom) role assignment CRUD operations back in. I would expect the final form of this PR to be only additions.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json (hunk 2 above)
Comment (Lusitanian) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json:36]: yes

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json (hunk 4 above)
Comment (Lusitanian) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json]: unclear to me what happened but it should be fixed now

**Review by vasanthrajams — APPROVED** [review]

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json [hunk 5]
```diff
@@ -365,6 +514,18 @@
...
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/RoleDefinitionProperties",
+          "description": "Role definition properties."
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "description": "Role assignment create parameters."
```
Comment (chlowell) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json:527]:
```suggestion
      "description": "Role definition create parameters."
```

**Comment by lmazuel** [comment]
Merging, approved by Alex offline

**Pull request merged**